core/xchain/server/server_test.go

```diff
@@ -551,6 +557,33 @@ func deployMockErc721Contract(
 	return auth, contractAddress, erc721
 }
 
+func TestLinkedWallets(t *testing.T) {
+	ctx, cancel := test.NewTestContext()
+	ctx = dlog.CtxWithLog(ctx, noColorLogger())
+	defer cancel()
+
+	require := require.New(t)
+	st := newServiceTester(5, require)
+	defer st.Close()
+	st.Start(t)
+
+	bc := st.ClientSimulatorBlockchain()
+	cfg := st.Config()
+	cs, err := client_simulator.New(ctx, cfg, bc, bc.Wallet)
+	require.NoError(err)
+	cs.Start(ctx)
+	defer cs.Stop()
+
+	auth, err := bind.NewKeyedTransactorWithChainID(anvilWallet.PrivateKeyStruct, big.NewInt(31337))
```

Comment:
In order to get wallet linking to work in your test case, you'll need to change all wallet linking calls to be called in this pattern - create an auth, assign a recent nonce, call the contract method, and wait for the txn to execute. You won't be able to use the TestBlockChainContext since it's configured to use the in-memory chain.